src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,114 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+    private T[] elements;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexForAdd(index);
+        ensureCapacity();
+        shiftRight(index);
```

Comment:
You don't create separate method shiftRight(index);
```suggestion
        System.arraycopy(elements, index, elements, index + 1, size - index);
```